### ISSUE TITLE
Array access breaking issue

### ISSUE THREAD

**Issue by jmorrell** [jlongster/prettier#136]
Input:
```js
const mostExpandedasdasdfasdfasdfasdf = doc.expandedStates[doc.expandedStates.length - 1];
```

Output:
```js
const mostExpandedasdasdfasdfasdfasdf = doc.expandedStates[doc.expandedStates.length -
  1];
```

Expected Output:
```js
const mostExpandedasdasdfasdfasdfasdf = doc.expandedStates[
  doc.expandedStates.length - 1
];
```

https://jlongster.github.io/prettier/#%7B%22content%22%3A%22const%20mostExpandedasdasdfasdfasdfasdf%20%3D%20doc.expandedStates%5Bdoc.expandedStates.length%20-%201%5D%3B%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3Afalse%2C%22bracketSpacing%22%3Atrue%7D%7D

If it's a function call, it's handled as expected:

https://jlongster.github.io/prettier/#%7B%22content%22%3A%22const%20mostExpandedasdasdfasdfasdfasdf%20%3D%20doc.expandedStates(doc.expandedStates.length%20-%201)%3B%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3Afalse%2C%22bracketSpacing%22%3Atrue%7D%7D

**Comment by vramana** [comment]
Would you expect
```js
const mostExpandedasdasdfasdfasdfasdf = doc.expandedStates["aLongLongPropertyName"].something();
```
to be 
```js
const mostExpandedasdasdfasdfasdfasdf = doc.expandedStates[
  "aLongLongPropertyName"
].something();
```

It makes sense to me. But is that what one would expect?

**Comment by jmorrell** [comment]
In that case I think I would expect what is currently done with function calls:

https://jlongster.github.io/prettier/#%7B%22content%22%3A%22const%20mostExpandedasdasdfasdfasdfasdf%20%3D%20doc.expandedStates(%5C%22aLongLongPropertyName%5C%22).something()%3B%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3Afalse%2C%22bracketSpacing%22%3Atrue%7D%7D

```js
const mostExpandedasdasdfasdfasdfasdf = doc
  .expandedStates["aLongLongPropertyName"]
  .something();
```

or for it to recognize that the array access can be rewritten as a property access in this case:

```js
const mostExpandedasdasdfasdfasdfasdf = doc
  .expandedStates
  .aLongLongPropertyName
  .something();
```

**Comment by vjeux** [comment]
Thanks for reporting this @jmorrell, I just sent a fix for the original problem: #471
